pkg/operation/botanist/containerruntime.go

```diff
@@ -39,85 +40,79 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-// DeployContainerRuntimeResources creates the `Container runtime` resource in the shoot namespace in the seed
-// cluster. Gardener waits until an external controller did reconcile the resources successfully.
+// DeployContainerRuntimeResources creates a `Container runtime` resource in the shoot namespace in the seed
+// cluster. Deploys one resource per CRI per Worker.
+// Gardener waits until an external controller has reconciled the resources successfully.
 func (b *Botanist) DeployContainerRuntimeResources(ctx context.Context) error {
 	fns := []flow.TaskFn{}
-	requiredContainerRuntimeTypes := sets.NewString()
 	for _, worker := range b.Shoot.Info.Spec.Provider.Workers {
 		if worker.CRI != nil {
 			for _, containerRuntime := range worker.CRI.ContainerRuntimes {
-				if !requiredContainerRuntimeTypes.Has(containerRuntime.Type) {
-
-					requiredContainerRuntimeTypes.Insert(containerRuntime.Type)
+				cr := containerRuntime
+				workerName := worker.Name
+				toApply := extensionsv1alpha1.ContainerRuntime{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      getContainerRuntimeKey(cr.Type, workerName),
+						Namespace: b.Shoot.SeedNamespace,
+					},
+				}
 
-					var (
-						cr      = containerRuntime
-						toApply = extensionsv1alpha1.ContainerRuntime{
-							ObjectMeta: metav1.ObjectMeta{
-								Name:      containerRuntime.Type,
-								Namespace: b.Shoot.SeedNamespace,
-							},
+				fns = append(fns, func(ctx context.Context) error {
+					_, err := controllerutil.CreateOrUpdate(ctx, b.K8sSeedClient.Client(), &toApply, func() error {
+						metav1.SetMetaDataAnnotation(&toApply.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
+						toApply.Spec.BinaryPath = extensionsv1alpha1.ContainerDRuntimeContainersBinFolder
+						toApply.Spec.Type = cr.Type
```

Comment:
doesn't this mean that we could have more than one container runtime per pool?

Reply:
It is designed to have one ContainerRuntime CRD per Pool per type. For instance fir worker with name "worker-1" having gVisor and Kata container runtime configured:  worker-1-gVisor & worker-1-kata

Reply:
okay got it, I just got confused from the description of the PR 

>Deploy one ContainerRuntime per Pool per CRI. 

I should have noticed the capital letter initials :D

Reply:
The description was confusing, sorry. Updated it :)